src/shared/components/home/emojis-form.tsx

```diff
@@ -0,0 +1,456 @@
+    private isoData = setIsoData(this.context);
+    private subscription: Subscription | undefined;
+    private itemsPerPage = 15;
+    private emptyState: EmojiFormState = {
+        loading: false,
+        siteRes: this.isoData.site_res,
+        customEmojis: this.isoData.site_res.custom_emojis.map((x, index) =>
+        ({
+            id: x.custom_emoji.id,
+            category: x.custom_emoji.category,
+            shortcode: x.custom_emoji.shortcode,
+            image_url: x.custom_emoji.image_url,
+            alt_text: x.custom_emoji.alt_text,
+            keywords: x.keywords.map(x => x.keyword).join(" "),
+            changed: false,
+            page: 1 + (Math.floor(index / this.itemsPerPage))
+        })),
+        page: 1
+    };
+    state: EmojiFormState;
+    private scrollRef: any = {};
+    constructor(props: any, context: any) {
+        super(props, context);
+        this.state = this.emptyState;
+
+        this.handlePageChange = this.handlePageChange.bind(this);
+        this.parseMessage = this.parseMessage.bind(this);
+        this.subscription = wsSubscribe(this.parseMessage);
+    }
+    get documentTitle(): string {
+        return i18n.t("custom_emojis");
+    }
+
+    componentWillUnmount() {
+        if (isBrowser()) {
+            this.subscription?.unsubscribe();
+        }
+    }
+
+    render() {
+        return (
+            <div className="col-12">
+                <HtmlTags
+                    title={this.documentTitle}
+                    path={this.context.router.route.match.url}
+                />
+                <h5 className="col-12">{i18n.t("custom_emojis")}</h5>
+                {customEmojisLookup.size > 0 && <div>
+                    <EmojiMart onEmojiClick={this.handleEmojiClick} pickerOptions={this.configurePicker()}></EmojiMart>
+                </div>}
+                <div className="table-responsive col-12">
+                    <table
+                        id="emojis_table"
+                        className="table table-hover"
+                    >
+                        <thead className="pointer">
+                            <tr>
+                                <th style="width:70px;">{i18n.t("column_emoji")}</th>
+                                <th style="width:150px">{i18n.t("column_shortcode")}</th>
+                                <th style="width:150px">{i18n.t("column_category")}</th>
+                                <th style="width:200px">{i18n.t("column_imageurl")}</th>
+                                <th style="width:150px">{i18n.t("column_alttext")}</th>
+                                <th>{i18n.t("column_keywords")}</th>
+                                <th style="width:121px"></th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                            {this.state.customEmojis.slice((this.state.page - 1) * this.itemsPerPage, ((this.state.page - 1) * this.itemsPerPage) + this.itemsPerPage)
```

Comment:
Leave complicated front-end paging logic out of this, if we need to later, we can add it in the back end.

Reply:
Regarding paging: All emojis are already loaded on frontend, no need to pull pages from server. We are paging on admin settings because showing all emojis at once (especially when you get to the thousands) causes page to slow to a crawl and become unusable.

Reply:
If we're really getting to the level of crashing the front end because there are too many emojis, then we really need to take them out of `GetSite` into a paged emoji fetcher. 

Front end hacks like this have to be replicated on all lemmy front ends, which means we're not structuring things correctly. I'm okay with merging this as is, but please add an issue to the Lemmy repo for breaking emojis into its own paged endpoint.

Reply:
I would consider adding a hardcoded emoji limit in the backend then. If you hit the limit, you just need to remove an existing emoji before adding a new one.

Reply:
All users need all of the emojis available; when they open the emoji-picker, we need to know what to put in there all at once, not paged. Because of this, its kind of pointless to also have paged endpoints for emojis. I personally consider front-end paging pretty trivial; its mainly necessary once you get to the number of emojis that sites like hexbear have, which ends up with >2000 <tr> elements. Rendering all those at once is unfortunately a weakness of html and will cause page to get unresponsive.

Adding a hard coded limit would be unfortunate, as hexbear would be way over the limit of what is responsive on that page. Seems unnecessary, since the obvious way to work with it is just paginating on the front end. Alternate frontends can go without paginating, as long as they understand that it will get slower as more emojis are added.